app/controllers/registrations_controller.rb

```diff
@@ -1,24 +1,25 @@
 class RegistrationsController < Devise::RegistrationsController
   def create
-    invite = verify_invite
-
+    verify_invite { return }
```

Comment:
If you decided to do it that way, then it's better to move this logic to before_action:

```ruby
before_action :verify_invite

def verify_invite
  unless ...
    return redirect_to new_user_registration_path, alert: 'Invalid token'\
  end
end
```
It shouldn't call an action if the redirect happens